src/Validate.Lib/Validators/EmailValidator.cs

```diff
@@ -0,0 +1,33 @@
+﻿
+using System;
+using System.Globalization;
+
+namespace FormatValidator.Validators
+{
+    internal class EmailValidator : ValidationEntry
+    {
+        public override bool IsValid(string toCheck)
+        {
+            bool isValid = true;
+            string code = "Invalid value: Field is not a valid email " +
+                "('@' not present, '.' not present or leading/trailing spaces present";
+
+            try
+            {
+                var addr = new System.Net.Mail.MailAddress(toCheck);
+                if (!addr.Address.Equals(toCheck))
```

Comment:
Is this check necessary? this if statement will never evaluate to false because the addr object is constructed with the toCheck string. Therefore they will always be equal.

Reply:
Now this one I had to find the best solution to and the `Address` property of the `MailAddress` object will not be the same as the address to check, in this case `toCheck`, if its invalid. If its syntactically invalid it will throw an exception.